sootup.core/src/main/java/sootup/core/validation/CheckInitValidator.java

```diff
@@ -31,18 +35,35 @@ public class CheckInitValidator implements BodyValidator {
   @Override
   public List<ValidationException> validate(Body body, View view) {
 
-    // TODO: #535 implement validator
-    //  check code copied from old soot
-    /*
-     * ExceptionalUnitGraph g = new ExceptionalUnitGraph(body, ThrowAnalysisFactory.checkInitThrowAnalysis(), false);
-     *
-     * InitAnalysis analysis = new InitAnalysis(g); for (Unit s : body.getUnits()) { FlowSet<Local> init =
-     * analysis.getFlowBefore(s); for (ValueBox vBox : s.getUseBoxes()) { Value v = vBox.getValue(); if (v instanceof Local)
-     * { Local l = (Local) v; if (!init.contains(l)) { throw new ValidationException(s,
-     * "Local variable $1 is not definitively defined at this point".replace("$1", l.getName()), "Warning: Local variable " +
-     * l + " not definitely defined at " + s + " in " + body.getMethod(), false); } } } }
-     */
-    return null;
+    List<ValidationException> validationException = new ArrayList<>();
+    List<String> predecessors = new ArrayList<>();
+    for (Stmt s : body.getStmts()) {
+      predecessors.add(s.toString());
```

Comment:
It's not good practice to work with strings like this. You can use actual Stmt objects.